src/descriptor/key.rs

```diff
@@ -58,6 +57,15 @@ pub enum DescriptorSecretKey {
     XPrv(DescriptorXKey<bip32::ExtendedPrivKey>),
 }
 
+/// A derived [`DescriptorPublicKey`]
+///
+/// Derived keys are guaranteed to never contain wildcards
+pub struct DerivedDescriptorKey<'secp, C: 'secp + Verification> {
+    key: DescriptorPublicKey,
+    index: u32,
+    secp: &'secp Secp256k1<C>,
```

Comment:
+1 for removing the `secp` since @sanket1729 says that it only takes 50 microseconds to create one and this will go away totally in the future (if I understand correctly).

Reply:
Initially, I meant it as a remark and was not suggesting a change to the PR.  

This comes at a cost that the derivation function would be twice slow. Would that be an acceptable cost to bdk? Curious what @apoelstra thinks about this. I don't think twice slow is a big deal, but I am not sure how bad it is for wallet re-scanning

Reply:
I don't think it would be too bad, we actually have a few other places where we can gain performances by caching stuff and we've never bothered doing so.

I would also wait for Andrew's opinion though, he knows secp and how to use it properly :)

Reply:
I like the idea of dropping the secp parameter. No opinion about whether this PR is the right time to do it. But we are moving upstream toward a model where verification contexts at least can be made basically for free.

Reply:
I think it's worth removing here since it removes the lifetime from the type which can get in the way depending on how you do things.